charts/dvpe-deployment-gloo/templates/virtualservice.yaml

```diff
@@ -16,6 +16,19 @@ spec:
       - {{ . | quote }}
       {{- end }}
     routes:
-      {{- toYaml .Values.gloo.virtualservice.spec.virtualHost.routes | nindent 6 }}
-
+      - matchers:
+        - prefix: {{ .Values.gloo.virtualservice.spec.virtualHost.routes.callbackUrlPath }}
+        routeAction:
+          single:
+            upstream:
+              name: {{ .Values.gloo.virtualservice.spec.virtualHost.routes.upstream.name }}
```

Comment:
take care here. this can be quite error prone since the upstream naming has a special syntax. this is described in the README you deleted :)
Instead of allowing to add a custom name, better build the upstream name based upon service/namespace name. you can create a template in _helpers.tpl